src/Bridge/ClientRepository.php

```diff
@@ -75,8 +75,6 @@ protected function handlesGrant($record, $grantType)
                 return $record->personal_access_client;
             case 'password':
                 return $record->password_client;
-            case 'client_credentials':
```

Comment:
This check was moved up to `validateClient` as the League server should only call `validateClient` for confidential clients, regardless of the grant type.